bibliopixel/layout/layout.py

```diff
@@ -12,19 +12,33 @@ def construct(cls, project, **desc):
         """Construct a layout."""
         return cls(project.drivers, maker=project.maker, **desc)
 
-    def __init__(self, drivers, threadedUpdate, brightness,
-                 maker=data_maker.MAKER, **kwds):
-        """Base LED class. Use Strip or Matrix instead!"""
+    def __init__(self, drivers, threadedUpdate=False, brightness=255,
+                 maker=data_maker.MAKER, color_list=None, **kwds):
+        """
+        Base LED class. Use Strip or Matrix instead!
+
+        Arguments
+            drivers -- a list of drivers
+            threadedUpdate -- if True, updates to this layout are done in a
+                separate thread
+            brightness - a brightness multiple from 0 to 255
+            maker - a data maker to make color_lists
+            color_list -- if non-Null, the layout uses this color_list instead
+                of creating its own
+        """
         attributes.set_reserved(self, 'layout', **kwds)
         self.drivers = drivers if isinstance(drivers, list) else [drivers]
         self.maker = maker
 
-        if not hasattr(self, 'numLEDs'):
-            self.numLEDs = sum(d.numLEDs for d in self.drivers)
-
-        # This buffer will always be the same list - i.e. is guaranteed to only
-        # be changed by list surgery, never assignment.
-        self._colors = maker.color_list(self.numLEDs)
+        # self._colors will always be the same list - i.e. is guaranteed only
+        # to be changed by list surgery, never assignment.
+        if color_list is None:
+            if not hasattr(self, 'numLEDs'):
+                self.numLEDs = sum(d.numLEDs for d in self.drivers)
+            self._colors = maker.color_list(self.numLEDs)
+        else:
+            self.numLEDs = len(color_list)
+            self._colors = color_list
```

Comment:
Ok, so now my question is why would you want this? Why would you ever send it it's own color_list?